reposcan/repodata/modules.py

```diff
@@ -22,39 +22,45 @@ def __init__(self, filename):
-                module_dict[name][stream]['version'] = int(data.get('version'))
-                module_dict[name][stream]['context'] = str(data.get('context'))
-                module_dict[name][stream]['stream'] = str(stream)
-                module_dict[name][stream]['profiles'] = data.get('profiles', [])
-                for profile_name in module_dict[name][stream]['profiles']:
-                    module_dict[name][stream]['profiles'][profile_name]['default_profile'] = False
+                if not stream in module_dict[name]:
+                    module_dict[name][stream] = []
+                new_stream = {}
+                new_stream['name'] = name
+                new_stream['arch'] = data.get('arch')
+                new_stream['version'] = int(data.get('version'))
+                new_stream['context'] = str(data.get('context'))
+                new_stream['stream'] = str(stream)
+                new_stream['profiles'] = data.get('profiles', [])
+                for profile_name in new_stream['profiles']:
+                    new_stream['profiles'][profile_name]['default_profile'] = False
                 if 'artifacts' in data:
-                    module_dict[name][stream]['artifacts'] = data.get('artifacts').get('rpms', [])
-                module_dict[name][stream]['default_stream'] = False
+                    new_stream['artifacts'] = data.get('artifacts').get('rpms', [])
+                new_stream['default_stream'] = False
+                module_dict[name][stream].append(new_stream)
             elif parsed['document'] == 'modulemd-defaults':
                 default_list.append(parsed)
         self._parse_modulemd_defaults(module_dict, default_list)
         for name in module_dict:
             for stream in module_dict[name]:
-                self.modules.append(module_dict[name][stream])
+                self.modules.extend(module_dict[name][stream])
 
     @staticmethod
     def _parse_modulemd_defaults(module_dict, default_list):
         for parsed in default_list:
             data = parsed['data']
             name = data.get('module')
             default_stream = data.get('stream')
-            if default_stream:
-                module_dict[name][default_stream]['default_stream'] = True
+            if default_stream and default_stream in module_dict[name]:
+                for stream_dict in module_dict[name][default_stream]:
+                    stream_dict['default_stream'] = True
             for stream in data['profiles']:
                 if stream not in module_dict[name]:
                     continue  # this is here due to module defaults referencing non-existing modules
                 default_profile = data['profiles'][stream]
-                if default_profile and default_profile[0] in module_dict[name][stream]['profiles']:
-                    module_dict[name][stream]['profiles'][default_profile[0]]['default_profile'] = True
+                if default_profile:
```

Comment:
This section is not covered by tests. Is it possible to fix it? Maybe by adding some data to testing file `modules.yaml`. As you can see everything but this section is covered: <https://codecov.io/gh/RedHatInsights/vmaas/src/d0eb311780953ab2d6f6ddc287620d225aed0f98/reposcan/repodata/modules.py>

Reply:
Otherwise I can merge it.

Reply:
This should be fixed now - https://codecov.io/gh/RedHatInsights/vmaas/src/a26bb0a099b9b0428b8f9ba8d69715a0b6f60864/reposcan/repodata/modules.py